src/projection.jl

```diff
@@ -262,19 +262,50 @@ end
 ##### `Base`, part II: return of the Tangent
 #####
 
+# Tuple
+function ProjectTo(xs::Tuple)
+    elements = map(ProjectTo, xs)
+    if elements isa Tuple{Vararg{ProjectTo{<:AbstractZero}}}
+        ProjectTo{NoTangent}()  # short-circuit if all elements project to zero
+    else
+        return ProjectTo{Tuple}(; type=Val(typeof(xs)), elements=elements)
+    end
+end
+
+_val(::Val{val}) where {val} = val
```

Comment:
Do you think it is worth explaining what this helper is for?
Also maybe we move it to bottom? Or top?

```suggestion
# often we need to push Types into type-params to prevent closures counting them as 
# DataType and then being unstable. So we use `Val` to push them up into type params
# and then _val to take them back out.
_val(::Val{val}) where {val} = val
```

Reply:
I moved to to the top, with a note.